File: packages/rust-cli/src/commands/logout.rs
use std::io::{self, Error, ErrorKind};

use dialoguer::Confirm;

use crate::auth::{get_token, rm_token};

pub fn logout() -> io::Result<()> {
    if let None = get_token()? {
        return Err(Error::new(ErrorKind::Other, "You are not logged in."));
    }

    match Confirm::new()
        .with_prompt("Are you sure you want to log out?")
        .interact()?
    {
        true => {
            rm_token()?;

            println!();
            println!("You have been logged out.");

            Ok(())
        }
        false => Err(Error::new(ErrorKind::Other, "Logout aborted.")),
    }
}

File: packages/old-cli/src/commands/logout.ts
import inquirer from 'inquirer';
import { deleteAuthFile, isLoggedIn } from '../auth';
import { logError, logSpace, logSuccess } from '../utils/logger';

export async function logout() {
  if (!isLoggedIn) {
    logError('You are not logged in.');
    return;
  }

  const { confirm } = await inquirer.prompt({
    type: 'confirm',
    name: 'confirm',
    message: 'Are you sure you want to log out?',
  });

  if (!confirm) {
    logError(`Log out aborted.`);
    return;
  }

  deleteAuthFile();

  logSpace();
  logSuccess('You have been logged out.');
}
